azure-iot-device/azure/iot/device/iothub_mqtt_client.py

```diff
@@ -106,6 +106,7 @@ async def generator() -> AsyncGenerator[_T, None]:
             async for mqtt_message in incoming_mqtt_messages:
                 try:
                     yield transform_fn(mqtt_message)
+                    mqtt_message = None
```

Comment:
It makes testing easier. The garbage collection will free it, but not until the generator stops running or the next message is received,  This is because the `mqtt_message` variable is set to the old message up until the point that the generator yields a new message.

Reply:
Strange - I had thought execution was suspended upon a yield until the next call to __anext__(), so I would assume this reset to None wouldn't happen right away.